specs/lib/compressTexturesSpec.js

```diff
@@ -0,0 +1,444 @@
+'use strict';
+var Cesium = require('cesium');
+var child_process = require('child_process');
+var clone = require('clone');
+var dataUriToBuffer = require('data-uri-to-buffer');
+var fsExtra = require('fs-extra');
+var path = require('path');
+var Promise = require('bluebird');
+var compressTextures = require('../../lib/compressTextures');
+var Pipeline = require('../../lib/Pipeline');
+
+var fsExtraReadJson = Promise.promisify(fsExtra.readJson);
+var fsExtraStat = Promise.promisify(fsExtra.stat);
+
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+
+var basePath = './specs/data/boxTexturedUnoptimized/';
+var gltfPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.gltf';
+var gltfEmbeddedPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTestEmbedded.gltf';
+
+// Defined relative to the gltf
+var jpgPath = 'Cesium_Logo_Flat.jpg';
+var jpegPath = 'Cesium_Logo_Flat.jpeg';
+var pngPath = 'Cesium_Logo_Flat.png';
+var gifPath = 'Cesium_Logo_Flat.gif';
+var decalPath = 'Cesium_Logo_Flat_Decal.png'; // Contains alpha channel
+
+function getCompressedTexture(gltfPath, imagePath, options) {
+    return fsExtraReadJson(gltfPath)
+        .then(function(gltf) {
+            if (defined(imagePath)) {
+                gltf.images.Image0001.uri = imagePath;
+            }
+            options.enable = true;
+            var pipelineOptions = {
+                textureCompressionOptions : options,
+                basePath : basePath
+            };
+            return Pipeline.processJSON(gltf, pipelineOptions)
+                .then(function(gltf) {
+                    return gltf.images.Image0001.uri;
+                });
+        });
+}
+
+function verifyKTX(gltfPath, imagePath, options, expectedFormat) {
+    return getCompressedTexture(gltfPath, imagePath, options)
```

Comment:
Should `getCompressedTexture` be named `compressTexture`, e.g., `doThis` instead of `getDoThis`?